signer/tests/integration/contracts.rs

```diff
@@ -1,3 +1,128 @@
+}
+
+pub struct SbtcDepositContract;
+
+impl AsContractDeploy for SbtcDepositContract {
+    const CONTRACT_NAME: &'static str = "sbtc-deposit";
+    const CONTRACT_BODY: &'static str = DEPOSIT;
+}
+
+pub struct SbtcWithdrawalContract;
+
+impl AsContractDeploy for SbtcWithdrawalContract {
+    const CONTRACT_NAME: &'static str = "sbtc-withdrawal";
+    const CONTRACT_BODY: &'static str = WITHDRAWAL;
+}
+
+pub struct SbtcBootstrapContract;
+
+impl AsContractDeploy for SbtcBootstrapContract {
+    const CONTRACT_NAME: &'static str = "sbtc-bootstrap-signers";
+    const CONTRACT_BODY: &'static str = BOOTSTRAP;
+}
+
+pub struct SignerKeyState {
+    pub state: SignerStxState,
+    pub keys: [Keypair; 3],
+}
+
+/// Deploy an sBTC smart contract to the stacks node
+async fn deploy_smart_contract<T>(state: &SignerKeyState, client: &StacksClient, deploy: T)
+where
+    T: AsContractDeploy,
+{
+    let mut unsigned = MultisigTx::new_tx(ContractDeploy(deploy), &state.state, TX_FEE);
+
+    let signatures: Vec<RecoverableSignature> = state
+        .keys
+        .iter()
+        .map(|kp| sign_ecdsa(unsigned.tx(), &kp.secret_key()))
+        .collect();
+
+    // This only fails when we are given an invalid signature.
+    for signature in signatures {
+        unsigned.add_signature(signature).unwrap();
+    }
+
+    let tx = unsigned.finalize_transaction();
+
+    match client.submit_tx(&tx).await.unwrap() {
+        SubmitTxResponse::Acceptance(_) => (),
+        SubmitTxResponse::Rejection(err) => println!("{}", serde_json::to_string(&err).unwrap()),
+    }
+}
+
+/// Deploy all sBTC smart contracts to the stacks node
+pub async fn deploy_smart_contracts() -> SignerKeyState {
+    static SBTC_DEPLOYMENT: OnceCell<bool> = OnceCell::const_new();
+    let (signer_wallet, key_pairs) = wallet::generate_wallet();
+
+    let settings = StacksSettings::new_from_config().unwrap();
+    let client = StacksClient::new(settings);
+
+    let account_info = client.get_account(&signer_wallet.address()).await.unwrap();
+    let state = SignerKeyState {
+        state: SignerStxState::new(signer_wallet, account_info.nonce),
+        keys: key_pairs,
+    };
+
+    SBTC_DEPLOYMENT
+        .get_or_init(|| async {
+            // The registry and token contracts need to be deployed first
+            // and second respectively. The rest can be deployed in any
+            // order.
+            deploy_smart_contract(&state, &client, SbtcRegistryContract).await;
+            deploy_smart_contract(&state, &client, SbtcTokenContract).await;
+            deploy_smart_contract(&state, &client, SbtcDepositContract).await;
+            deploy_smart_contract(&state, &client, SbtcWithdrawalContract).await;
+            deploy_smart_contract(&state, &client, SbtcBootstrapContract).await;
+            true
+        })
+        .await;
+
+    state
+}
+
+#[ignore]
+#[tokio::test]
+async fn test_deploy() {
+    let _state = deploy_smart_contracts().await;
```

Comment:
How do we know if the deployment has been successful? I don't see any assertion being made anywhere.

Reply:
Oh yeah, I was supposed to remove this test. It's only here so that I can test that I setup smart contract deployment correctly.

Reply:
Oh and to silence the warnings. I'm going to keep it in for now and remove it when I get to https://github.com/stacks-network/sbtc/issues/264.

Reply:
Alright :+1: so this PR is basically only for the utilities and not the actual tests.

Reply:
Yup, just for setup utilities.